spec/models/claim/verify_response_parameters_parser_spec.rb

```diff
@@ -0,0 +1,271 @@
+require "rails_helper"
+
+RSpec.describe Claim::VerifyResponseParametersParser do
+  describe "#gender" do
+    it "returns :male when Verify reports 'gender' as 'MALE'" do
+      parser = Claim::VerifyResponseParametersParser.new(sample_response_parameters(gender: {value: "MALE", verified: true}))
+      expect(parser.gender).to eq :male
+    end
+
+    it "returns :female when Verify reports 'gender' as 'FEMALE'" do
+      parser = Claim::VerifyResponseParametersParser.new(sample_response_parameters(gender: {value: "FEMALE", verified: true}))
+      expect(parser.gender).to eq :female
+    end
+
+    it "returns nil when Verify reports 'gender' as neither 'MALE' nor 'FEMALE" do
+      parser = Claim::VerifyResponseParametersParser.new(sample_response_parameters(gender: {value: "OTHER", verified: true}))
+      expect(parser.gender).to be_nil
+    end
+
+    it "returns a gender, even when the value is not 'verified'" do
+      parser = Claim::VerifyResponseParametersParser.new(sample_response_parameters(gender: {value: "MALE", verified: false}))
+      expect(parser.gender).to eq :male
+    end
+
+    it "returns nil if the 'gender' is not reported at all" do
+      parser = Claim::VerifyResponseParametersParser.new(sample_response_parameters.except("gender"))
+      expect(parser.gender).to be_nil
+    end
+  end
+
+  describe "#date_of_birth" do
+    it "returns the raw String value for the 'datesOfBirth' reported by Verify" do
+      parser = Claim::VerifyResponseParametersParser.new(sample_response_parameters(datesOfBirth: [{value: "1994-08-09"}]))
+      expect(parser.date_of_birth).to eq "1994-08-09"
+    end
+  end
+
+  describe "#first_name" do
+    it "returns a 'verified' value for 'firstNames'" do
+      parser = Claim::VerifyResponseParametersParser.new(sample_response_parameters(firstNames: [{value: "Bob", verified: true}]))
+      expect(parser.first_name).to eq "Bob"
+    end
+
+    it "returns the most recent 'verified' value for 'firstNames'" do
+      changed_name_parameters = sample_response_parameters({
+        firstNames: [
+          {value: "Barbara", verified: true, from: "1991-12-12", to: "2018-08-08"},
+          {value: "Bob", verified: true, from: "2018-08-09"},
+        ],
+      })
+      parser = Claim::VerifyResponseParametersParser.new(changed_name_parameters)
+
+      expect(parser.first_name).to eq "Bob"
+    end
+
+    it "returns the 'verified' value without a from/to date (assuming this will be the most recent)" do
```

Comment:
Should we assume this? What if the `to` date is not populated? Which one do we trust more?

Reply:
I think we can only go off the Verify responses we've seen, which essentially look like the data structure below: one with both a from/to, one with neither.

Reply:
The fact that one has a `to` in the past would suggest it isn't current.

Reply:
True. Best not to make assumptions about the state of the data, especially as this is precisely why we have a private beta 👍

Reply:
should I update the wording of the spec to remove the word "assuming"?

Reply:
Yeah, please. Might save confusion further down the line

Reply:
done.